src/rules/package-lavamoat-tsup-conforms.ts

```diff
@@ -0,0 +1,48 @@
+import { buildRule } from './build-rule';
+import { RuleName } from './types';
+
+export default buildRule({
+  name: RuleName.PackageLavamoatTsupConform,
+  description:
+    'Does the `lavamoat.allowscripts` field in `package.json` conform?',
+  dependencies: [RuleName.RequireValidPackageManifest],
+  execute: async ({ project, template, pass, fail }) => {
+    const entryPath = 'package.json';
+    const templateManifest = await template.fs.readJsonFile(entryPath);
+    const projectManifest = await project.fs.readJsonFile(entryPath);
+
+    type Key = keyof typeof templateManifest;
+    const templateLavamoat = templateManifest?.['lavamoat' as Key];
+    const projectLavamoat = projectManifest?.['lavamoat' as Key];
+    if (templateLavamoat && projectLavamoat) {
+      type ChildKey = keyof typeof templateLavamoat;
```

Comment:
Is this something that `dataConforms` can help with?

Reply:
Yeah, that's a good point, perhaps we can't use it here because of that special case. No worries if not.